### PR TITLE
feat/ custom font size

### PR DESCRIPTION
**Pull request by talevy17**

### CODE REVIEW

File: packages/common/web/src/types/editorCommandsType.ts [hunk 1]
```diff
@@ -110,6 +110,7 @@ export interface EditorCommands {
     pluginsIncluded: string[];
   };
   getColor: (colorType: ColorType) => string | undefined;
+  getFontSize: () => number | undefined;
```
Comment (shaulgo) [inline packages/common/web/src/types/editorCommandsType.ts:113]: window.getComputedStyle(document.getElementById("foo")).fontSize;

File: packages/ricos-content/web/src/converters/draft/fromDraft/getTextNodes.ts [hunk 2]
```diff
@@ -32,6 +32,30 @@ const mergeColorDecorations = (decorations: Decoration[]): Decoration[] => {
 const isDecorationType = (decorationType: string) =>
   TO_RICOS_DECORATION_TYPE[decorationType] !== undefined;
 
+const isComplexDecorationType = (decorationType: string) =>
+  ['FG', 'BG', 'font-size'].includes(decorationType);
+
+const getComplexDecoration = {
```
Comment (shaulgo) [inline packages/ricos-content/web/src/converters/draft/fromDraft/getTextNodes.ts:38]: dynamicDecorationGetters

File: packages/ricos-content/web/src/converters/draft/fromDraft/getTextNodes.ts [hunk 3]
```diff
@@ -32,6 +32,30 @@ const mergeColorDecorations = (decorations: Decoration[]): Decoration[] => {
 const isDecorationType = (decorationType: string) =>
   TO_RICOS_DECORATION_TYPE[decorationType] !== undefined;
 
+const isComplexDecorationType = (decorationType: string) =>
+  ['FG', 'BG', 'font-size'].includes(decorationType);
```
Comment (shaulgo) [inline packages/ricos-content/web/src/converters/draft/fromDraft/getTextNodes.ts:36]:
```suggestion
```

File: packages/ricos-content/web/src/converters/draft/fromDraft/getTextNodes.ts [hunk 4]
```diff
@@ -65,13 +89,10 @@ export const getTextNodes = (block: RicosContentBlock, entityMap: RicosEntityMap
       const styleObj = JSON.parse(style);
       const type = Object.keys(styleObj)[0];
       const value = Object.values<string>(styleObj)[0];
-      if (type !== 'FG' && type !== 'BG') {
+      if (!isComplexDecorationType(type)) {
```
Comment (shaulgo) [inline packages/ricos-content/web/src/converters/draft/fromDraft/getTextNodes.ts:92]: delete schema validation

File: packages/ricos-content/web/src/converters/draft/fromDraft/getTextNodes.ts [hunk 5]
```diff
@@ -32,6 +32,30 @@ const mergeColorDecorations = (decorations: Decoration[]): Decoration[] => {
...
+  BG: (value: string) => {
+    return {
+      type: Decoration_Type.COLOR,
+      colorData: { background: value },
+    };
+  },
+  'font-size': (value: string) => {
+    return {
+      type: Decoration_Type.FONTSIZE,
+      fontSize: value,
```
Comment (shaulgo) [inline packages/ricos-content/web/src/converters/draft/fromDraft/getTextNodes.ts:54]: let's think if we want to save as an object. Maybe we will want more then px units

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: packages/editor/web/src/RichContentEditor/utils/fontSizeUtils.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  getSelectedBlocks,
+  hasOneStyleInSelection,
+} from 'wix-rich-content-editor-common';
+import { uniq } from 'lodash';
+import { EditorProps } from 'draft-js';
+import { ThemeData } from 'wix-rich-content-common';
+
+// Temporarily taken from common/statics/styles/consts.scss should replace with themed\docStyle consts when supported
+const defaultFontSizes = {
+  'header-one': '32',
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/utils/fontSizeUtils.ts:14]: move defaults to theme

File: packages/editor/web/src/RichContentEditor/utils/fontSizeUtils.ts [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  'header-two': 'h2',
+  'header-three': 'h3',
+  'header-four': 'h4',
+  'header-five': 'h5',
+  'header-six': 'h6',
+  unstyled: 'p',
+};
+
+const getBlockFontSizeByType = (type: string, themeData?: ThemeData) => {
+  return themeData?.customStyles?.[draftToRicosCustomStyles[type]] || defaultFontSizes[type];
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/utils/fontSizeUtils.ts:34]: get font size

File: packages/editor/web/src/RichContentEditor/EditorCommands.ts [hunk 8]
```diff
@@ -223,6 +231,7 @@ export const createEditorCommands = (
     },
     getAnchorableBlocks: () => getAnchorableBlocks(getEditorState()),
     getColor: colorType => getColor(getEditorState(), colorType),
+    getFontSize: () => getFontSize(getEditorState(), themeData),
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/EditorCommands.ts:234]: No need to pass all the theme

File: packages/editor/web/src/RichContentEditor/utils/fontSizeUtils.ts [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+export const setFontSize = (editorState: EditorState, data?: { fontSize?: string }) => {
+  const selection = editorState.getSelection();
+  const newEditorState = removeCurrentFontSize(editorState);
+  let contentState = newEditorState.getCurrentContent();
+  if (data?.fontSize) {
+    const inlineStyle = JSON.stringify({ 'font-size': data?.fontSize + 'px' });
+    contentState = Modifier.applyInlineStyle(contentState, selection, inlineStyle);
+  }
+  return EditorState.push(newEditorState, contentState, 'change-inline-style');
+};
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/utils/fontSizeUtils.ts:88]: use from draftUtils

File: packages/viewer/web/src/inlineStyleMapper.tsx [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+import { InlineStyleMapper, getBlocksFromContentState } from 'wix-rich-content-common';
+
+const parseStyle = style => {
+  try {
+    const rule = JSON.parse(style);
+    return rule['font-size'] ? rule : undefined;
+  } catch (e) {
+    return undefined;
+  }
+};
```
Comment (shaulgo) [inline packages/viewer/web/src/inlineStyleMapper.tsx:11]: add util safeJsonParse

File: packages/viewer/web/src/inlineStyleMapper.tsx [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+const fontSizeInlineStyleMapper = raw => {
+  const rawBlocks = getBlocksFromContentState(raw);
+  const mapper = rawBlocks.reduce<InlineStyleMapper>((map, block) => {
+    if (block.inlineStyleRanges) {
+      block.inlineStyleRanges
+        .filter(range => !!parseStyle(range.style))
```
Comment (shaulgo) [inline packages/viewer/web/src/inlineStyleMapper.tsx:18]: !!safeJsonParse(range.style)?.['font-size']

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: packages/editor/web/src/RichContentEditor/utils/fontSizeUtils.ts (hunk 7 above)
Comment (talevy17) [reply packages/editor/web/src/RichContentEditor/utils/fontSizeUtils.ts]: done

File: packages/editor/web/src/RichContentEditor/EditorCommands.ts (hunk 8 above)
Comment (talevy17) [reply packages/editor/web/src/RichContentEditor/EditorCommands.ts]: passing only customStyles

File: packages/editor/web/src/RichContentEditor/utils/fontSizeUtils.ts (hunk 9 above)
Comment (talevy17) [reply packages/editor/web/src/RichContentEditor/utils/fontSizeUtils.ts]: moved to `inlineStyleUtils`

File: packages/viewer/web/src/inlineStyleMapper.tsx (hunk 10 above)
Comment (talevy17) [reply packages/viewer/web/src/inlineStyleMapper.tsx]: done

File: packages/viewer/web/src/inlineStyleMapper.tsx (hunk 11 above)
Comment (talevy17) [reply packages/viewer/web/src/inlineStyleMapper.tsx]: done

File: packages/editor/web/src/RichContentEditor/utils/fontSizeUtils.ts (hunk 6 above)
Comment (talevy17) [reply packages/editor/web/src/RichContentEditor/utils/fontSizeUtils.ts]: @Barackos thinks that adding the defaults to `customStyles` abuses the idea of `customStyles` (i.e defaults are not custom) - injecting the defaults with `CSSProperties` is possible but not worth the effort according to him

File: packages/editor-common/web/src/Utils/inlineStyleUtils.js [hunk 12]
```diff
@@ -46,3 +60,23 @@ export const getSelectionStyles = (styleSelectionPredicate, editorState) => {
     }, [])
   );
 };
+
+export const removeCurrentInlineStyle = (editorState, styleSelectionPredicate) => {
```
Comment (shaulgo) [inline packages/editor-common/web/src/Utils/inlineStyleUtils.js:64]: check if can be with color

File: packages/editor/web/src/RichContentEditor/EditorCommands.ts [hunk 13]
```diff
@@ -93,9 +93,13 @@ import {
   RICOS_LINK_TYPE,
   RICOS_MENTION_TYPE,
   RICOS_CODE_BLOCK_TYPE,
+  RICOS_FONT_SIZE_TYPE,
   UNSUPPORTED_BLOCKS_TYPE,
+  RicosCustomStyles,
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/EditorCommands.ts:98]: remove

File: packages/editor/web/src/RichContentEditor/EditorCommands.ts [hunk 14]
```diff
@@ -318,7 +326,7 @@ export const createEditorCommands = (
       }
     },
     triggerDecoration: type => {
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/EditorCommands.ts:328]: check if 
```suggestion
    triggerDecoration: (type: string) => {
```

File: packages/viewer/web/src/inlineStyleMapper.tsx [hunk 15]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import {
+  InlineStyleMapper,
+  getBlocksFromContentState,
+  safeJsonParse,
+} from 'wix-rich-content-common';
+
+const fontSizeInlineStyleMapper = raw => {
```
Comment (shaulgo) [inline packages/viewer/web/src/inlineStyleMapper.tsx:8]: add color

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: packages/common/web/src/Utils/viewerUtils.tsx [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+  RICOS_FONT_SIZE_TYPE,
+} from 'ricos-content';
+import { isHexColor } from './colorUtils';
+
+export type CustomInlineStyleType =
+  | typeof RICOS_TEXT_COLOR_TYPE
+  | typeof RICOS_TEXT_HIGHLIGHT_TYPE
+  | typeof RICOS_FONT_SIZE_TYPE;
+
+export const parseStyleByType = (style: string, type: CustomInlineStyleType) => {
```
Comment (shaulgo) [inline packages/common/web/src/Utils/viewerUtils.tsx:14]: `keyof INLINE_STYLE_TYPES` should also work. So either use it here or use it to define the type. But if you still define the type make sure to have them next to each other.

File: packages/common/web/src/Utils/viewerUtils.tsx [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+  return cssRule?.[property];
+};
+
+export const INLINE_STYLE_TYPES = {
+  [RICOS_TEXT_COLOR_TYPE]: 'FG',
+  [RICOS_TEXT_HIGHLIGHT_TYPE]: 'BG',
+  [RICOS_FONT_SIZE_TYPE]: 'font-size',
+};
+
+export const getDynamicStyleParser = {
```
Comment (shaulgo) [inline packages/common/web/src/Utils/viewerUtils.tsx:26]: if you make a function you name it with a verb. If it's an object let's name it dynamicStyleParsers

File: packages/ricos-common/web/src/pluginsStrategy/pluginsStrategy.spec.ts [hunk 18]
```diff
@@ -69,6 +69,6 @@ describe('PluginsStrategy', () => {
       'themeData',
     ]);
     expect(result.typeMappers.length).toEqual(2);
-    expect(result.inlineStyleMappers.length).toEqual(1);
+    expect(result.inlineStyleMappers.length).toEqual(0);
```
Comment (shaulgo) [inline packages/ricos-common/web/src/pluginsStrategy/pluginsStrategy.spec.ts:72]: remove

File: e2e/tests/__snapshots__/plugins.e2e.ts.snap [hunk 19]
```diff
@@ -0,0 +1,374 @@
+exports[`plugins > spoiler > check text spoilers in editor and reveal it in viewer #0`] =
```
Comment (shaulgo) [inline e2e/tests/__snapshots__/plugins.e2e.ts.snap:1]: delete

**Review by shaulgo — APPROVED** [review]

**Pull request merged**